src/renderer/components/cluster-manager/clusters-menu.tsx

```diff
@@ -57,7 +57,11 @@ export class ClustersMenu extends React.Component<Props> {
       menu.append(new MenuItem({
         label: _i18n._(t`Disconnect`),
         click: async () => {
+          const lastActiveClusterId = workspaceStore.currentWorkspace.lastActiveClusterId;
           if (clusterStore.isActive(cluster.id)) {
+            if (lastActiveClusterId === cluster.id) {
+              workspaceStore.setLastActiveClusterId("");
```

Comment:
This is very weird, `""` is not a valid `ClusterId` (so some function might throw if this gets out). There should be a better way to represent this (`undefined` maybe given that `lastActiveClusterId` is optional.

Reply:
Good catch! Refactored this part.